pythonforandroid/bootstraps/common/build/src/main/java/org/kivy/android/DashCamPythonService.java

```diff
@@ -0,0 +1,212 @@
+package org.kivy.android;
+
+import android.os.Build;
+import java.lang.reflect.Method;
+import java.lang.reflect.InvocationTargetException;
+import android.app.Service;
+import android.os.IBinder;
+import android.os.Bundle;
+import android.content.Intent;
+import android.content.Context;
+import android.util.Log;
+import android.app.Notification;
+import android.app.PendingIntent;
+import android.os.Process;
+import java.io.File;
+
+//imports for channel definition
+import android.app.NotificationManager;
+import android.app.NotificationChannel;
+import android.graphics.Color;
+
+public class DashCamPythonService extends Service implements Runnable {
+
+    private static final String SERVICE_ENTRY_POINT= "gen3cam/services/dashCamUploaderService.py";
+    private static final String PYTHON_NAME = "dashCamUploaderService";
+
+    // Thread for Python code
+    private Thread pythonThread = null;
+
+    // Python environment variables
+    private String androidPrivate;
+    private String androidArgument;
+    private String pythonName;
+    private String pythonHome;
+    private String pythonPath;
+    private String serviceEntrypoint;
+    // Argument to pass to Python code,
+    private String pythonServiceArgument;
+
+
+    public static DashCamPythonService mService = null;
+    private Intent startIntent = null;
+
+    private boolean autoRestartService = false;
+
+    public void setAutoRestartService(boolean restart) {
+        autoRestartService = restart;
+    }
+
+    public int startType() {
+        return START_NOT_STICKY;
+    }
+
+    @Override
+    public IBinder onBind(Intent arg0) {
+        return null;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        if (pythonThread != null) {
+            Log.v("python service", "service exists, do not start again");
+            return START_NOT_STICKY;
+        }
+
+		startIntent = intent;
+        if ( intent != null) {
+            Bundle extras = intent.getExtras();
+            androidPrivate = extras.getString("androidPrivate");
+            androidArgument = extras.getString("androidArgument");
+            serviceEntrypoint = extras.getString("serviceEntrypoint");
+            pythonName = extras.getString("pythonName");
+            pythonHome = extras.getString("pythonHome");
+            pythonPath = extras.getString("pythonPath");
+            boolean serviceStartAsForeground = (
+                extras.getString("serviceStartAsForeground").equals("true")
+            );
+            pythonServiceArgument = extras.getString("pythonServiceArgument");
+            pythonThread = new Thread(this);
```

Comment:
Where is `pythonThread` used? I can see the thread is started but work is not submitted.

Reply:
the workload is the native execution (that is the python entry point).

Reply:
Where is referenced?

Reply:
Ignore I clicked, as the service impements `Runnable`, the run method starts the python code